File: src/main/java/classifications/Spell.java
package java.classifications;

public class Spell extends EntryRef{
    
    private String spellname;
    private String school;
    private String duration;
    private String components;
    private int level;
    private String range;
    private String castingTime;
    private String description;
    private String ingredients;
    private boolean ritual;
    private boolean concentration;

    public String getName() {
        return spellname;
    }

    public void setName(String spellname) {
        this.spellname = spellname;
    }

    public String getRange() {
        return range;
    }

    public void setRange(String range) {
        this.range = range;
    }

    public String getTime(){
        return castingTime;
    }

    public void setTime(String castingTime){
        this.castingTime = castingTime;
    }
    
    public String getDescription(){
        return description;
    }

    public void setDescription(String description){
        this.description = description;
    }

    public Boolean isRitual(){
        return ritual;
    }

    public void setRitual(Boolean ritual){
        this.ritual = ritual;
    }

    public String getIngredients(){
        return ingredients;
    }

    public void setIngredients(String ingredients){
        this.ingredients = ingredients;
    }

       public String getSchool() {
        return school;
    }

    public void setSchool(String school) {
        this.school = school;
    }

     public String getDuration() {
        return duration;
    }

    public void setDuration(String duration) {
        this.duration = duration;
    }

    public String getComponents() {
        return components;
    }

    public void setComponents(String components) {
        this.components = components;
    }

    public int getLevel() {
        return level;
    }

    public void setLevel(int level) {
        this.level = level;
    }

    public void setConcentration(boolean concentration) {
        this.concentration = concentration;
    }
    
    public boolean getConcentration() {
        return concentration;
    }
}

File: src/main/kotlin/idiotamspielen/vttproject/models/Spell.kt
package idiotamspielen.vttproject.models

import com.fasterxml.jackson.annotation.JsonAlias
import com.fasterxml.jackson.annotation.JsonProperty

data class Spell(
    @JsonProperty("name") @JsonAlias("spellName") override val name: String,
    @JsonProperty("school") val school: String,
    @JsonProperty("duration") val duration: String,
    @JsonProperty("components") val components: String,
    @JsonProperty("level") val level: Int,
    @JsonProperty("range") val range: String,
    @JsonProperty("castingTime") val castingTime: String,
    @JsonProperty("description") val description: String,
    @JsonProperty("ingredients") val ingredients: String,
    @JsonProperty("ritual") val ritual: Boolean,
    @JsonProperty("concentration") val concentration: Boolean
) : Nameable
